server/src/main/java/com/defold/extender/Extender.java

```diff
@@ -862,51 +862,75 @@ else if (platform.contains("osx")) {
         mergedContextWithPodsForObjC.put("systemIncludes", new ArrayList<String>());
         mergedContextWithPodsForObjCpp.put("systemIncludes", new ArrayList<String>());
         mergedContextWithPodsForSwift.put("systemIncludes", new ArrayList<String>());
-
+        // also clearing systemIncludes for c
+        // see https://github.com/defold/extender/issues/389
+        mergedContextWithPodsForC.put("systemIncludes", new ArrayList<String>());
 
         List<String> objs = new ArrayList<>();
 
         if (!pod.swiftSourceFiles.isEmpty()) {
+            // Add -enable-experimental-feature AccessLevelOnImport
+            // There are a lot of extensions with Swift code that seem to require this
+            // flag. Perhaps swiftFlags should be added as a new field in ext.manifest?
+            // https://github.com/swiftlang/swift-evolution/blob/main/proposals/0409-access-level-on-imports.md
+            List<String> swiftFlags = (List<String>)mergedContextWithPodsForSwift.get("swiftFlags");
+            List<String> updatedSwiftFlags = new ArrayList<String>(swiftFlags);
+            updatedSwiftFlags.add("-enable-experimental-feature");
+            updatedSwiftFlags.add("AccessLevelOnImport");
+            mergedContextWithPodsForSwift.put("swiftFlags", updatedSwiftFlags);
+
             // generate headers from swift files
             List<String> emitSwiftHeaderCommands = new ArrayList<>();
-            emitSwiftHeaders(pod, mergedContextWithPodsForC, emitSwiftHeaderCommands);
+            LOGGER.info("emit swift header");
+            emitSwiftHeader(pod, mergedContextWithPodsForSwift, emitSwiftHeaderCommands);
             ProcessExecutor.executeCommands(processExecutor, emitSwiftHeaderCommands); // in parallel
 
             // generate swift module from swift files
             List<String> emitSwiftModuleCommands = new ArrayList<>();
-            emitSwiftModule(pod, mergedContextWithPodsForC, emitSwiftModuleCommands);
+            LOGGER.info("emit swift module");
+            emitSwiftModule(pod, mergedContextWithPodsForSwift, emitSwiftModuleCommands);
             ProcessExecutor.executeCommands(processExecutor, emitSwiftModuleCommands); // in parallel
 
             // compile swift source files one by one
             List<String> compileSwiftCommands = new ArrayList<>();
             for (File src : pod.swiftSourceFiles) {
-                final int i = getAndIncreaseNameCount();
-                File o = addCompileFileSwift(pod, i, src, mergedContextWithPodsForC, compileSwiftCommands);
-                objs.add(ExtenderUtil.getRelativePath(jobDirectory, o));
+                if (!podSourceLookup.contains(src)) {
```

Comment:
Some pods with multiple subspecs have source file mappings covering the same source files. This will result in duplicate symbols when linking. We avoid this by tracking which files we have already compiled.